packages/client/__tests__/GoogleAuthentication.test.tsx

```diff
@@ -0,0 +1,560 @@
+                                            });
+                                        });
+                                    },
+                                },
+                            },
+                            'Unknown server error': {
+                                meta: {
+                                    test: async () => {
+                                        await waitFor(() => {
+                                            expect(
+                                                Toast.show,
+                                            ).toHaveBeenCalledWith({
+                                                type: 'error',
+                                                text1: 'Continue with google error',
+                                                text2: 'We encountered an error please try again later',
+                                            });
+                                        });
+                                    },
+                                },
+                            },
+                        },
+                    },
+                },
+
+                on: {
+                    'Make google and server send back successful oauth': {
+                        target: '#app.Rendering home screen',
+                    },
+                    'Make google authentication fail by cancel': {
+                        target: 'Rendering signing in screen.Google authentication failed.User cancelled google oauth',
+                    },
+                    'Make google authentication fail by dismiss': {
+                        target: 'Rendering signing in screen.Google authentication failed.User dismissed google oauth',
+                    },
+                    'Make google authentication fail by locked': {
+                        target: 'Rendering signing in screen.Google authentication failed.User google account is locked',
+                    },
+                    'Make google authentication fail by response error': {
+                        target: 'Rendering signing in screen.Google authentication failed.Google response error',
+                    },
+                    'Make server send back an nickname nor email unavailable error':
+                        {
+                            target: 'Rendering signing in screen.Server user access token verification error.Unavailable related nickname or email',
+                        },
+                    'Make server send back an nickname nor email invalid error':
+                        {
+                            target: 'Rendering signing in screen.Server user access token verification error.Invalid related nickname or email',
+                        },
+                    'Make server send back an unknown error': {
+                        target: 'Rendering signing in screen.Server user access token verification error.Unknown server error',
+                    },
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏When dealing with events in `@xstate/test` that modify the setup, I prefer to dissociate them from actions that a user could actually do.

It means that I would not call `pressContinueWithGoogleButton()` in the same events that modify how the server responds. To keep tracks in the machine that the setup has changed, I would assign to the context a boolean indicating that, like here: https://github.com/AdonisEnProvence/MusicRoom/blob/f38450da3a4fa7924bbd3696ffec3709683963a7/packages/client/__tests__/AuthenticationPasswordReset.test.tsx#L39.

But, to be honest, there is a lot of possible cases. And writing all these booleans, and a guard for each, would be tedious!

Reply:
Indeed you're completely right !
As you say a lot to deal with I went to the fast solution